### PR TITLE
Added comments to messages in localisation

### PR DESCRIPTION
**Pull request by KipHamiltons**
Most of these are probably wrong lol

### CODE REVIEW

File: shared/message/localisation/ResetRobotHypotheses.proto [hunk 1]
```diff
@@ -23,14 +23,30 @@ package message.localisation;
...
+
     message Self {
-        vec2   position     = 1;
-        mat2   position_cov = 2;
-        double heading      = 3;
-        double heading_var  = 4;
-        bool   absoluteYaw  = 5;
+
+        /**
+         * Estimate of the robot's {x, y} position.
```
Comment (Bidski) [inline shared/message/localisation/ResetRobotHypotheses.proto:34]: This is probably in field space?

Comment (KipHamiltons) [reply shared/message/localisation/ResetRobotHypotheses.proto:34]: judging by this passage: 
```cpp
            ResetRobotHypotheses::Self leftSide;
            // Start on goal line
            leftSide.position << -fieldDescription.dimensions.field_length * 0.5,
                fieldDescription.dimensions.field_width / 2;
```
I think you're right. I'll add that to the comment.

Comment (TrentHouliston) [reply shared/message/localisation/ResetRobotHypotheses.proto]: This is one of the main reasons why we have been trying to change our variable names to the form rSFf

**Review by ElderNoSpace — APPROVED** [review]
LGTM

**Review by Bidski — APPROVED** [review]
LGTM I guess

**Comment by KipHamiltons** [comment]
@TrentHouliston any objections? you didn't seem too keen [last time this was brought up](https://github.com/NUbots/NUbots/pull/374#issuecomment-783743484), but this seems pretty neat to me 🤷

**Pull request merged**